pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -235,6 +218,33 @@ func (m *kubeGenericRuntimeManager) configureContainerSwapResources(lcr *runtime
 	}
 }
 
+// getPodSwapBehavior checks what swap behavior should be configured for the pod, considering the
+// requirements for enabling swap.
+func (m *kubeGenericRuntimeManager) GetPodSwapBehavior(pod *v1.Pod, container *v1.Container) kubelettypes.SwapBehavior {
+	c := kubelettypes.SwapBehavior(m.memorySwapBehavior)
+	if c == kubelettypes.LimitedSwap {
+		if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeSwap) || !swapControllerAvailable() {
+			return kubelettypes.NoSwap
+		}
+
+		if !isCgroup2UnifiedMode() {
+			return kubelettypes.NoSwap
+		}
+
+		if kubelettypes.IsCriticalPod(pod) {
+			return kubelettypes.NoSwap
+		}
+		podQos := kubeapiqos.GetPodQOS(pod)
+		containerDoesNotRequestMemory := container.Resources.Requests.Memory().IsZero() && container.Resources.Limits.Memory().IsZero()
+		memoryRequestEqualsToLimit := container.Resources.Requests.Memory().Cmp(*container.Resources.Limits.Memory()) == 0
+		if podQos != v1.PodQOSBurstable || containerDoesNotRequestMemory || !isCgroup2UnifiedMode() || memoryRequestEqualsToLimit {
```

Comment:
`isCgroup2UnifiedMode` is already checked above
```suggestion
		if podQos != v1.PodQOSBurstable || containerDoesNotRequestMemory || memoryRequestEqualsToLimit {
```